### PR TITLE
chore: Allow setting compiler flags at the start of a file

### LINKED ISSUE #544: Refactor compilation mode syntax
Currently, we set special compilation modes via a comment directive at the start of a file, i.e.:

```
/* compilation-mode: runtime */
```

We should do two things:
1. Refactor compilation modes to work via a compiler flag, and
2. Add the ability to set compiler flags at the start of a file.

Not sure exactly what this should look like, but probably something like this:

```
/* grainc-flags --compilation-mode=runtime */
```

### PR DESCRIPTION
**Pull request by ospencer**
Closes #544.

@phated Why yes, I _did_ stay up until 4am getting this done.

### CODE REVIEW

File: compiler/src/utils/config.re [hunk 1]
```diff
@@ -268,6 +244,56 @@ let with_cli_options = (term: 'a): Cmdliner.Term.t('a) => {
   folded;
 };
 
+let with_unapplied_cli_options = (term: 'a): Cmdliner.Term.t('a) => {
```
Comment (ospencer) [inline compiler/src/utils/config.re:247]: I don't love having to need this, but it was the only way I could get `with_used_args` to be populated.

File: compiler/src/utils/config.re [hunk 2]
```diff
@@ -311,6 +337,22 @@ let principal =
     false,
   );
 
+let compilation_mode =
+  opt(
+    ~names=["compilation-mode"],
+    ~conv=
+      option_conv(
+        Cmdliner.Arg.enum([
+          ("normal", "normal"),
```
Comment (phated) [inline compiler/src/utils/config.re:346]: Can cmdliner do something like this?
```suggestion
          ("normal", Env.normal),
```

**Review by phated — APPROVED** [review]
Just 1 question that might be informative of future cleanup, but this looks **great**!

File: compiler/src/utils/config.re (hunk 2 above)
Comment (ospencer) [reply compiler/src/utils/config.re:346]: Yes, but `Env` depends on `Config`. We could potentially move it into some separate file that they can both depend on, though.

**Pull request merged**